### PR TITLE
[OPS-1110] Add instance: Sadalbari

### PR DESCRIPTION
**Pull request by mkaito**
This instance runs  [`serokell/hackage-search`](https://github.com/serokell/hackage-search).

I'm not really sure about the instance size. The cache currently takes up 5.8Gb of space on Jupiter. I believe the search service is pretty lean, but the download service uses a fair bit of CPU/RAM.

Service PR: https://github.com/serokell/hackage-search/pull/15

@int-index what do you think is the smallest possible amount of RAM I can give a server that will run `hackage-download`?

### CODE REVIEW

**Comment by int-index** [comment]
`hackage-search` relies on Linux loading its data set into RAM, or else it will work but very slowly. So 5GiB for the data set and 2GiB on top sounds about right.

As to the download service, it is non-critical, so let it use the swap space I guess

**Comment by mkaito** [comment]
That's more RAM than youtrack uses, jesus man.

**Comment by int-index** [comment]
It can function with less RAM, but the queries will take longer (also depends on the storage speed). We can try 2GiB and see how it performs.

**Comment by mkaito** [comment]
A CX31 has 8Gb of RAM. We're currently still evaluating Hetzner Cloud in terms of reliability and performance. Since hackage-search is both regularly used and not actually mission critical, I suppose it makes for a good test app.

File: flake.nix [hunk 1]
```diff
@@ -9,6 +9,13 @@
       flake = false;
     };
     deploy-rs.url = "github:serokell/deploy-rs";
+    hackage-search = {
+      type = "github";
+      owner = "serokell";
+      repo = "hackage-search";
+      ref = "mkaito/ops1110-flake";
```
Comment (balsoft) [inline flake.nix:16]: Don't forget to change before merging. Otherwise, LGTM for now

**Review by zhenyavinogradov — APPROVED** [review]
Looks good, just minor comments

File: servers/sadalbari/hackage-search.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+let profile = "/nix/var/nix/profiles/per-user/deploy/hackage-search";
+in {
+  services.hackage-search = {
+    enable = true;
+    package = profile;
+  };
+
+  # Do not attempt to start this unit unless the app profile path exists
+  systemd.services.hackage-search.unitConfig.ConditionPathExists = [ profile ];
```
[the remark this answers is not in the document]
Comment (mkaito) [reply servers/sadalbari/hackage-search.nix]: Removed

File: servers/sadalbari/hackage-search.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  # Allow the user to restart the backend service for CD
+  security.sudo.extraRules = let
+    rule = cmd: {
+      users = [ "deploy" ];
+      commands = [{
+        command = "/run/current-system/sw/bin/systemctl restart ${cmd}";
+        options = [ "NOPASSWD" ];
+      }];
+    };
+    in builtins.map rule [ "hackage-search" "hackage-download" ];
```
[the remark this answers is not in the document]
Comment (mkaito) [reply servers/sadalbari/hackage-search.nix]: Removed

**Review by balsoft — APPROVED** [review]

**Comment by balsoft** [comment]
Don't forget to change the reference.

**Review by balsoft — APPROVED** [review]

**Pull request merged**